src/promptflow/promptflow/_sdk/_serving/monitor/flow_monitor.py

```diff
@@ -0,0 +1,83 @@
+# ---------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# ---------------------------------------------------------
+
+import time
+from promptflow._sdk._serving.monitor.data_collector import FlowDataCollector
+from promptflow._sdk._serving.monitor.streaming_monitor import StreamingMonitor
+from promptflow._sdk._serving.monitor.metrics import MetricsRecorder, ResponseType
+from promptflow._sdk._serving.utils import streaming_response_required, get_cost_up_to_now
+from promptflow._sdk._serving.flow_result import FlowResult
+from promptflow._utils.exception_utils import ErrorResponse
+from flask import request, g
+
+
+class FlowMonitor:
```

Comment:
How can 3p customers get use of this monitor? Any doc on this?

Reply:
It will not in this PR, this one is just to migrate all cloud serving without any feature change. 
There will be separate PR for how to expose the monitoring feature to 3p customers.